### ISSUE TITLE
[3.4.5] port-forward connection breaks if user navigates using non-SSL protocol

### ISSUE THREAD

**Issue by mahmoudajawad** [argoproj/argo-workflows#10719]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

While the docs explains server UI needs to be accessed over `https://localhost:2746`, I usually assume all services will do SSL redirection if necessary, and thus, I attempted to navigate in browser by only typing `localhost:2746`, which reported a connection error in browser and following in console:
```
$ kubectl -n argo port-forward deployment/argo-server 2746:2746
Forwarding from 127.0.0.1:2746 -> 2746                                                                                    
Forwarding from [::1]:2746 -> 2746                                                                                        
Handling connection for 2746                                                                                              
E0321 08:20:00.779908   42303 portforward.go:406] an error occurred forwarding 2746 -> 2746: error forwarding port 2746 to
 pod ID, uid : network namespace for sandbox "ID" is closed                                                             
E0321 08:20:00.780423   42303 portforward.go:234] lost connection to pod 
```

After minutes of working around the issue, I finally checked server pod logs and found line:
```
time="2023-03-21T08:19:43.249Z" level=info msg="Argo Server started successfully on https://localhost:2746" url="https://localhost:2746"
```
which encouraged me to try accessing the server over SSL protocol myself, which worked.

Expected behaviour:
 - Server UI should not fail if accessed without SSL, causing port-forwarding to fail.
 - Server UI should do redirection from non-SSL to SSL connection.

### Version

v.3.4.5

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
N/A
```

### Logs from the workflow controller

```text
N/A
```

### Logs from in your workflow's wait container

```text
N/A
```

**Comment by tico24** [comment]
You can of course set `--secure=false` on the argo server deployment and use http if you wish.

https://argoproj.github.io/argo-workflows/upgrading/#defbd600e-fix-default-argo_securetrue-fixes-5607-5626

**Comment by mahmoudajawad** [comment]
@tico24, agree. I was just not expecting the behaviour itself. It was confusing to me why is port-forward connection is closing while the pod logs are not showing fatal error records.
